File: solutions/cpp/space-age/1/space_age.cpp
#include "space_age.h"

namespace space_age
{

    space_age::space_age(uint64_t seconds) : seconds_{seconds} {};

    uint64_t space_age::seconds() const
    {
        return seconds_;
    };

    double space_age::on_earth() const
    {
        return (double)seconds_ / EARTH_YEAR_SECS;
    };
    double space_age::on_mercury() const
    {
        return (double)seconds_ / MERCURY_YEAR_SECS;
    };
    double space_age::on_venus() const
    {
        return (double)seconds_ / VENUS_YEAR_SECS;
    };
    double space_age::on_mars() const
    {
        return (double)seconds_ / MARS_YEAR_SECS;
    }
    double space_age::on_jupiter() const
    {
        return (double)seconds_ / JUPITER_YEAR_SECS;
    };
    double space_age::on_saturn() const
    {
        return (double)seconds_ / SATURN_YEAR_SECS;
    };
    double space_age::on_uranus() const
    {
        return (double)seconds_ / URANUS_YEAR_SECS;
    };
    double space_age::on_neptune() const
    {
        return (double)seconds_ / NEPTUNE_YEAR_SECS;
    };

} // namespace space_age

File: solutions/java/space-age/1/src/main/java/SpaceAge.java
class SpaceAge {

    private static double EARTH_YEAR_IN_SECS = 31557600f;
    private double earthAge;

    SpaceAge(double seconds) {
        this.earthAge = (seconds / EARTH_YEAR_IN_SECS);
    }

    double onEarth() {
        return this.earthAge;
    }

    double onMercury() {
        final double mercuryYearToEarthYear = 0.2408467;
        return this.earthAge / mercuryYearToEarthYear;
    }

    double onVenus() {
        final double venusYearToEarthYear = 0.61519726;
        return this.earthAge / venusYearToEarthYear;
    }

    double onMars() {
        final double marsYearToEarthYear = 1.8808158;
        return this.earthAge / marsYearToEarthYear;
    }

    double onJupiter() {
        final double jupiterYearToEarthYear = 11.862615;
        return this.earthAge / jupiterYearToEarthYear;
    }

    double onSaturn() {
        final double saturnYearToEarthYear = 29.447498;
        return this.earthAge / saturnYearToEarthYear;
    }

    double onUranus() {
        final double uranusYearToEarthYear = 84.016846;
        return this.earthAge / uranusYearToEarthYear;
    }

    double onNeptune() {
        final double neptuneYearToEarthYear = 164.79132;
        return this.earthAge / neptuneYearToEarthYear;
    }

}
